modules/doctor/fix8312.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package doctor
+
+import (
+	"context"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/models/db"
+	org_model "code.gitea.io/gitea/models/organization"
+	"code.gitea.io/gitea/models/perm"
+	"code.gitea.io/gitea/modules/log"
+	"xorm.io/builder"
+)
+
+func fixIncorrectCreateOrgRepoPermission(ctx context.Context, logger log.Logger, autofix bool) error {
+	count := 0
+
+	err := db.Iterate(
+		ctx,
+		builder.Eq{"authorize": perm.AccessModeOwner, "can_create_org_repo": false},
+		func(ctx context.Context, team *org_model.Team) error {
+			team.CanCreateOrgRepo = true
+			count++
+
+			if !autofix {
+				return nil
+			}
+
+			return models.UpdateTeam(team, false, false)
+		},
+	)
+	if err != nil {
+		logger.Critical("Unable to iterate across repounits to fix incorrect can_create_org_repo: Error %v", err)
+		return err
+	}
+
+	if !autofix {
+		if count == 0 {
+			logger.Info("Found no team with incorrect can_create_org_repo")
+		} else {
+			logger.Warn("Found %d teams with incorrect can_create_org_repo", count)
+		}
```

Comment:
```suggestion
		} 
			logger.Warn("Found %d teams with incorrect can_create_org_repo", count)
		
```

Reply:
Quick question : shall we go with update migration too.

Reply:
@lunny 
This issue has a long history, maybe gitea doctor is enough?

Reply:
@puni9869 
I can not understand what this suggestion for. Can you add some descriptions?

Reply:
Maybe both doctor and migration are better and doctor could be backport to 1.20